### ISSUE TITLE
SMTP Client tutorial uses bad markup

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4200]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)| @khorn reported|
|-|-|
|Trac ID|trac#4200|
|Type|enhancement|
|Created|2010-01-05 19:21:32Z|
|Branch|https://github.com/twisted/twisted/tree/smtpclient-markup-4200|

The SMTP client tutorial in doc/mail/tutorial/smtpclient/smtpclient.xhtml uses some really nasty markup for its code samples.  It uses contructs like:

```
<blockquote><code>
EXAMPLE CODE HERE
</code></blockquote>
```

when it should use

```
<pre>
EXAMPLE CODE HERE
</pre>
```

This borks up Lore's handling of code samples (syntax highlighting, etc.), and is just generally nasty.

Attachments:

* [4200_bad_markup_fix.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c79/c79fce75b1583ddd36a96178757e0d8d0ac91228/baa75f6ac8483e78ec98d889ec770e3c1c7b39ff.patch) (14924 bytes) - added by khorn on 2010-01-05 19:34:46Z - Patch to fix bad markup.
* [4200_bad_markup_fix.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c79/c79fce75b1583ddd36a96178757e0d8d0ac91228/e5abb01de8c6e67d62c12e21ba54d83c5b31db76.patch) (15917 bytes) - added by khorn on 2010-01-05 21:41:49Z - New patch to address review comments.

<details><summary>Searchable metadata</summary>

```
trac-id__4200 4200
type__enhancement enhancement
reporter__khorn khorn
priority__normal normal
milestone__ 
branch__branches_smtpclient_markup_4200 branches/smtpclient-markup-4200
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__documentation documentation
time__1262719292000000 1262719292000000
changetime__1262748614000000 1262748614000000
version__None None
owner__ 
cc__khorn cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_1"></a>|@khorn removed owner|
|-|-|

Patch attached.  Sending to review.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess set owner to @khorn|
|-|-|

Your changes look good and lore fine. While this file is being worked on:
 * Can the 'Stuff' header get changed to something more helpful? 'Introduction' or 'Building an SMTP Client' would be better (although the whole document is structured weirdly by having the examples and conclusion live under the intro.
 * XXX's in howtos don't inspire confidence for users. Can the empty Conclusion section and XXX comment get commented out.
 * Deferred Execution Howto links back to the smtp tutorial instead of the deferred document.
 * spellchecking yielded only issueing => issuing

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_5"></a>|@khorn commented|
|-|-|

Replying to [jesstess](#note_4):
> Your changes look good and lore fine. While this file is being worked on:
>  * Can the 'Stuff' header get changed to something more helpful? 'Introduction' or 'Building an SMTP Client' would be better (although the whole document is structured weirdly by having the examples and conclusion live under the intro.

Done.

>  * XXX's in howtos don't inspire confidence for users. Can the empty Conclusion section and XXX comment get commented out.

Done.

>  * Deferred Execution Howto links back to the smtp tutorial instead of the deferred document.

Done.

>  * spellchecking yielded only issueing => issuing

Done.

New patch on the way.

Probably the whole thing needs to be revamped, but let's do that in a separate ticket.

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_6"></a>|@khorn removed owner|
|-|-|

Back to review...

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess commented|
|-|-|

(In [27911]) Branching to 'smtpclient-markup-4200'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_9"></a>|@jesstess commented|
|-|-|

I changed the deferred hyperlink from ```<a href="../../core/howto/defer.html">``` to ```<a href="../../../core/howto/defer.html">```. Other than that, looks good! Merging.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess commented|
|-|-|

(In [27912]) Apply 4200_bad_markup_fix.2.patch, fix some linewrapping, fix
Deferreds hyperlink.

refs [#4200](https://github.com/twisted/twisted/issues/4200)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_11"></a>|@jesstess set status to `closed`|
|-|-|

(In [27913]) Merge smtpclient-markup-4200

Author: khorn
Reviewer: jesstess
Fixes: [#4200](https://github.com/twisted/twisted/issues/4200)

Clean up markup around code snippets, fix a hyperlink, and
miscellaneous other cosmetic touchups.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
